src/types.jl

```diff
@@ -315,10 +317,25 @@ function System(;
     L = typeof(loggers)
     F = typeof(force_units)
     E = typeof(energy_units)
-    return System{D, G, T, A, AD, PI, SI, GI, C, V, B, NF, L, F, E}(
+
+    if energy_units==NoUnits
+        if unit(k)==NoUnits #Use user-supplied unitless Boltzmann constant
+            k_converted=k
+        else
+            k_converted=ustrip(u"u* nm^2 * ps^-2 * K^-1",k) #otherwise assume energy units are (u* nm^2 * ps^-2)
```

Comment:
Wrap this in `Zygote.ignore() do` ... `end` to make tests pass (this will be removed later once some gradient support in the Unitful ecosystem is tagged).